web_src/js/features/repo-common.js

```diff
@@ -96,6 +101,75 @@ export function initRepoCloneLink() {
   });
 }
 
+export function initCitationFileCopyContent() {
+  const defaultCitationFormat = 'apa'; // apa or bibtex
+
+  const $citationCopyApa = $('#citation-copy-apa');
+  const $citationCopyBibtex = $('#citation-copy-bibtex');
+  const $inputContent = $('#citation-copy-content');
+
+  if ((!$citationCopyApa.length && !$citationCopyBibtex.length) || !$inputContent.length) {
+    return;
+  }
+
+  const initInputCitationValue = () => {
+    const {fileContent} = pageData;
+    const config = plugins.config.get('@bibtex');
+    config.constants.fieldTypes.doi = ['field', 'literal'];
+    config.constants.fieldTypes.version = ['field', 'literal'];
+    const citationFormatter = new Cite(fileContent);
+    const apaOutput = citationFormatter.format('bibliography', {
+      template: 'apa',
+      lang: 'en-US'
+    });
+    const bibtexOutput = citationFormatter.format('bibtex', {
+      lang: 'en-US'
+    });
+    $citationCopyBibtex.attr('data-text', bibtexOutput);
+    $citationCopyApa.attr('data-text', apaOutput);
+  };
+  initInputCitationValue();
+
+  const updateUi = () => {
+    let isBibtex = (localStorage.getItem('citation-copy-format') || defaultCitationFormat) === 'bibtex';
+    if (isBibtex && $citationCopyApa.length === 0) {
```

Comment:
I think if `$citationCopyApa` is empty, you actually want to use bibitext. But is there a reason of why it could be empty?